pkg/virt-handler/non-root.go

```diff
@@ -122,6 +122,32 @@ func (d *VirtualMachineController) prepareTap(vmi *v1.VirtualMachineInstance, re
 
 }
 
+func (*VirtualMachineController) prepareVFIO(vmi *v1.VirtualMachineInstance, res isolation.IsolationResult) error {
+	vfioPath := filepath.Join(res.MountRoot(), "dev", "vfio")
+	err := os.Chmod(filepath.Join(vfioPath, "vfio"), 0666)
```

Comment:
I dont think we need to change the permissions, when vfio module is loaded it changes the permissions to 666.

Reply:
Interesting, can you point me to some documentation about this?

Reply:
@ormergi ping

Reply:
@xpivarc sorry for this late response, I didn't find where I read it yet.
But I verified it on my local setup , when you load vfio kernel module the permissions are changed.

Reply:
I believe this is not system agnostic. For example, my local setup has these permissions:
```
ls /dev/vfio/vfio -al
crw-------. 1 root root 10, 196 Nov 24 14:32 /dev/vfio/vfio
```
This change is needed otherwise it wouldn't fail on sriov lane in the first place. The good news is here https://kubernetes.io/blog/2021/11/09/non-root-containers-and-devices/ but we need to be able to consume this change on all supported versions.

Reply:
Interesting, what happen to `/dev/vfio/vfio` permissions after `modprobe vfio`?

You mean this PR is needed unless SRIOV lane wont pass, right?
I am raising this just to see if there is another way around other than mass changing permissions to `/dev/vfio` and its groups.
After looking at this PR few times I realized that you only change the vfio groups that being use by Kubevirt, which is great!

(Regarding the blog post you mention I didn't read it yet)

Reply:
Well according to [vfio kernel docs](https://www.kernel.org/doc/html/latest/driver-api/vfio.html?highlight=vfio#vfio-usage-example) it suggested to set the permissions to /dev/vfio/vfio as you did,
also it is not mentioned that vfio will change /dev/vfio/vfio permissions.

Reply:
Note that sriov-network-operator and non of its components touching /dev/vfio/* permissions.

Reply:
Note that sriov-network-operator and non of its components touching /dev/vfio/* permissions.

Reply:
Note that sriov-network-operator and non of its components touching /dev/vfio/* permissions.

Reply:
We are only changing it for containers and we don't touch host. This is a general problem of device plugins. The permissions of the devices passed to the container are not altered(they just copy what the host has). Therefore non-root users will fail to use the devices.

Reply:
I see thanks for explaining :slightly_smiling_face:
I thought it mounts the original device (so when you change it inside the container it is reflected on the host).
Could you please refer me to the docs that explain this if there is one?

Reply:
I am not aware of the docs that explain this all in one. But you can simply verify my point by doing ssh to kubevirtci nodes and checking `kvm` device. Then start a VM and have a look there and back on the node.

Reply:
> I see thanks for explaining slightly_smiling_face I thought it mounts the original device (so when you change it inside the container it is reflected on the host). Could you please refer me to the docs that explain this if there is one?

@ormergi You can follow [this](https://github.com/opencontainers/runc/blob/00f56786bb220b55b41748231880ba0e6380519a/libcontainer/rootfs_linux.go#L694) part of the code which will hopefully give you the right picture. Feel free to ping me if you would like to learn more.